packages/radio-group/src/RadioGroup.tsx

```diff
@@ -75,17 +75,20 @@ function RadioGroup({
   value: controlledValue,
   name: nameProp,
 }: RadioGroupProps) {
-  const isControlled = controlledValue != null;
+  let isControlled = controlledValue != null ? true : false,
```

Comment:
Previously strictly determining if a Radio is controlled based on the value prop. Since we're already iterating through the Radios to find a defaultChecked, makes sense to search for the `checked` prop, which will also let us know that the component is controlled.